### PR TITLE
Bus: Remove unused wildcard handlers and clean up tests

### PR DESCRIPTION
**Pull request by marcusolsson**
**What this PR does / why we need it**:
This removes the wildcard listeners, as the functionality doesn't seem to be used anywhere. Please correct me if I'm wrong on this.

Also splits a test and adds a few more.

### CODE REVIEW

File: pkg/bus/bus_test.go [hunk 1]
```diff
@@ -13,82 +13,118 @@ type testQuery struct {
...
 		return nil
-	}
+	})
 
-	handler := func(query *testQuery) error {
-		handlerCallCount++
-		return nil
-	}
+	err := bus.Dispatch(&testQuery{})
+	require.Nil(t, err)
```
Comment (aknuds1) [inline pkg/bus/bus_test.go]: Maybe you can use `require.NoError` for checking the absence of errors? Doesn't make much difference, but should make eventual failures more explicit.

**Review by aknuds1 — APPROVED** [review]
I don't know the code from before, but I don't see any problems. I just have the comment that you might want to use `require.NoError` in the tests.

**Review by marefr — APPROVED** [review]
Agree with Arve. Otherwise LGTM

**Review by torkelo — APPROVED** [review]
Yea, your right it does look like it's used!

**Comment by marcusolsson** [comment]
Thanks for the heads-up on NoError. Definitely need to check out testify in more detail.

**Pull request merged**